src/bun.js/bindings/NodeHTTP.cpp

```diff
@@ -132,13 +132,12 @@ static void assignHeadersFromUWebSocketsForCall(uWS::HttpRequest* request, JSVal
         size++;
     }
 
-    JSC::JSObject* headersObject = JSC::constructEmptyObject(globalObject, globalObject->objectPrototype(), std::min(size, static_cast<size_t>(JSFinalObject::maxInlineCapacity)));
-    RETURN_IF_EXCEPTION(scope, void());
     JSC::JSArray* setCookiesHeaderArray = nullptr;
     JSC::JSString* setCookiesHeaderString = nullptr;
     MarkedArgumentBuffer arrayValues;
 
-    args.append(headersObject);
+    HashMap<RefPtr<UniquedStringImpl>, JSValue, IdentifierRepHash, HashTraits<RefPtr<UniquedStringImpl>>> headersMap;
```

Comment:
This is a performance hit in a hot code path (every single response in node:http's server).

Instead, we can read over the code in putDirect to figure out where it decides what the offset is in the structure and then use that to decide if it already has the property defined (removing the need for this hash table).